app/src/main/java/com/nutomic/syncthingandroid/activities/FolderPickerActivity.java

```diff
@@ -156,13 +156,13 @@ private void populateRoots() {
     @Override
     public void onServiceConnected(ComponentName componentName, IBinder iBinder) {
         super.onServiceConnected(componentName, iBinder);
-        getService().registerOnApiChangeListener(this);
+        getService().registerOnServiceStateChangeListener(this);
     }
 
     @Override
     protected void onDestroy() {
         super.onDestroy();
-        getService().unregisterOnApiChangeListener(this);
+        getService().unregisterOnServiceStateChangeListener(this);
     }
```

Comment:
Can this be null?

Reply:
No can't be, same explanation as I posted in https://github.com/syncthing/syncthing-android/pull/1119#discussion_r193385835 (see DeviceActivity question).
We should anyway null check this to make future implementations more easy not to run into issues here. Done by commit https://github.com/Catfriend1/syncthing-android/commit/9393e6fdd8869c93a95ad16fb76bb1fcd7b3ba0b .